tools/imatrix/imatrix.cpp

```diff
@@ -22,31 +24,49 @@
 static void print_usage(int, char ** argv) {
     LOG("\nexample usage:\n");
     LOG("\n    %s \\\n"
-            "       -m model.gguf -f some-text.txt [-o imatrix.dat] [--process-output] \\\n"
+            "       -m model.gguf -f some-text.txt [-o imatrix.gguf] [--process-output] \\\n"
             "       [--no-ppl] [--chunk 123] [--output-frequency 10] [--save-frequency 0] \\\n"
-            "       [--in-file imatrix-prev-0.dat --in-file imatrix-prev-1.dat ...] \\\n"
+            "       [--in-file imatrix-prev-0.gguf --in-file imatrix-prev-1.gguf ...] \\\n"
             "       [--parse-special]\n" , argv[0]);
     LOG("\n");
 }
 
+static bool str_has_suffix(const std::string & str, const std::string & suffix) {
+    return str.size() >= suffix.size() && str.compare(str.size() - suffix.size(), str.size(), suffix) == 0;
+}
+
+static bool str_remove_suffix(std::string & str, const std::string & suffix) {
+    bool has_suffix = str_has_suffix(str, suffix);
+    if (has_suffix) {
+        str = str.substr(0, str.size() - suffix.size());
+    }
+    return has_suffix;
+}
```

Comment:
```suggestion
static bool str_remove_suffix(std::string & str, const std::string_view & suffix) {
    bool has_suffix = string_ends_with(str, suffix);
    if (has_suffix) {
        str = str.substr(0, str.size() - suffix.size());
    }
    return has_suffix;
}
```
This is a nice complement to `string_ends_with`, should be moved to `common.cpp` as `string_remove_suffix`.

Reply:
I did wonder about that, but didn't know enough of the details behind it, reading that it certainly seems so, and it should be worth going over the usage of `std::string_view` elsewhere.